lib/engine-addon.js

```diff
@@ -294,17 +294,22 @@ module.exports = {
         return buildExternalTree.call(this);
       };
 
+
       this.import = function(asset, options) {
         var host = originalFindHost.call(this);
         var target = this._findHost();
 
+        target.env = host.env;
         target._import = host._import;
         target._getAssetPath = host._getAssetPath;
         target.otherAssets = host.otherAssets;
         target._mergeTrees = host._mergeTrees;
 
         // We're delegating to the upstream EmberApp behavior for eager engines.
         if (this.lazyLoading !== true) {
+          // Asset can be an object with environment properties.
+          asset = target._getAssetPath(asset);
+
```

Comment:
I think `asset` can be `undefined` here depending on `env`, so we should guard for that and return early.